lib/plugins/rest.js

```diff
@@ -189,23 +189,62 @@ exports.register = function (server, options, next) {
 						uri      : request.payload.uri,
 						revision : request.payload.revision
 					});
-
-					return instances.updateInstance(instance)
-					.then(function (response) {
-						reply(response);
-					})
-					.catch (function (error) {
-						server.log([ "error", "rest" ], error.message);
-						if (error.toString().match(/\bnot exist\b/)) {
-							reply(Boom.notFound("This instance does not exist."));
-						}
-						else if (error.toString().match(/\bconcurrency\b/)) {
-							reply(Boom.conflict("Conflict error. Ensure you have the latest document version"));
+					switch (instance.state) {
+						case ("terminating") : {
+							return instances.updateInstance(instance)
+							.then(function (response) {
+								reply(response);
+								return awsAdapter.terminateInstances(response.id);
+							})
+							.then(function () {
+								return instances.getInstance({ id : request.params.id });
+							})
+							.then(function (response) {
+								instance = new Instance({
+									id       : response.id,
+									ami      : response.ami,
+									type     : response.type,
+									state    : "terminated",
+									uri      : null,
+									revision : response.revision
+								});
+								return instances.updateInstance(instance);
+							})
+							.catch (function (error) {
```

Comment:
I think the way I have it worked out right now is that the error cases only reply when `updateInstance()` fails the first time. The rest of the work happens in the background and that's what the `else` block is for. I'll verify this to make sure this is actually the case.